azure/functions/_thirdparty/typing_inspect.py

```diff
@@ -0,0 +1,375 @@
+# Imported from https://github.com/ilevkivskyi/typing_inspect/blob/168fa6f7c5c55f720ce6282727211cf4cf6368f6/typing_inspect.py
+# Author: Ivan Levkivskyi
+# License: MIT
+
+"""Defines experimental API for runtime inspection of types defined
+in the standard "typing" module.
+
+Example usage::
+    from typing_inspect import is_generic_type
+"""
+
+# NOTE: This module must support Python 2.7 in addition to Python 3.x
+
+import sys
+NEW_TYPING = sys.version_info[:3] >= (3, 7, 0)  # PEP 560
+if NEW_TYPING:
+    import collections.abc
+
+
+if NEW_TYPING:
+    from typing import (
+        Generic, Callable, Union, TypeVar, ClassVar, Tuple, _GenericAlias
+    )
+else:
+    from typing import (
+        Callable, CallableMeta, Union, _Union, TupleMeta, TypeVar,
+        _ClassVar, GenericMeta,
+    )
+
+
+# from mypy_extensions import _TypedDictMeta
+
+
+def _gorg(cls):
+    """This function exists for compatibility with old typing versions."""
+    assert isinstance(cls, GenericMeta)
+    if hasattr(cls, '_gorg'):
+        return cls._gorg
+    while cls.__origin__ is not None:
+        cls = cls.__origin__
+    return cls
+
+
```

Comment:
This looks like a general utility. Shoudn't it be part of the python worker?

Reply:
It already is, but I'd like to avoid adding a dependency on the worker.